app/controllers/PaypalController.scala

```diff
@@ -154,18 +154,20 @@ class PaypalController(paymentServices: PaymentServices, corsConfig: CorsConfig,
 
     def okResult(payment: Payment): Result = {
       val response = render {
-        case Accepts.Json() => Ok(JsNull)
+        case Accepts.Json() => {
```

Comment:
you could write this as

```scala
case Accepts.Json() if supportRedirect.contains(true) => Ok(Json.obj("email" -> payment.getPayer.getPayerInfo.getEmail))
case Accepts.Json() => Ok(JsNull)
```

Just a stylistic point though so I'll leave it up to you to decide which you think is nicer

Reply:
I prefer your style, I'll go for it